File: src/main/java/ru/zakazsharovekb/airdelivery/model/Order.java
package ru.zakazsharovekb.airdelivery.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;
import ru.zakazsharovekb.airdelivery.common.enums.OrderStatus;
import ru.zakazsharovekb.airdelivery.common.enums.PayStatus;
import ru.zakazsharovekb.airdelivery.common.enums.ReceivingType;
import ru.zakazsharovekb.airdelivery.common.enums.TransferType;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Embedded
    private Customer customer;
    @JsonFormat(pattern = "dd/MM/yyyy HH:mm:ss")
    private LocalDateTime dateCreated;
    @Enumerated(EnumType.STRING)
    private TransferType transferType;
    @Embedded
    private Delivery delivery;
    @Enumerated(EnumType.STRING)
    private OrderStatus status;
    private String orderComment;
    @Enumerated(EnumType.STRING)
    private PayStatus payStatus;

    public Order() {
        this.customer = new Customer();
        this.delivery = new Delivery();
    }

    @Enumerated(EnumType.STRING)
    private ReceivingType receivingType;
    private double sum;
}

File: delivery_mobile_app/lib/order.dart
import 'customer.dart';
import 'delivery.dart';

class Order {
  int id;
  Customer customer;
  String dateCreated;
  String transferType;
  Delivery delivery;
  String status;
  String orderComment;
  String payStatus;
  String receivingType;
  double sum;

  Order(
      {required this.id,
      required this.customer,
      required this.dateCreated,
      required this.transferType,
      required this.delivery,
      required this.status,
      required this.orderComment,
      required this.payStatus,
      required this.receivingType,
      required this.sum});

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
        id: json['id'],
        customer: Customer.fromJson(json['customer']),
        dateCreated: json['dateCreated'] ?? "",
        transferType: json['transferType'] ?? "",
        delivery: Delivery.fromJson(json['delivery']),
        status: json['status'] ?? "",
        orderComment: json['orderComment'] ?? "",
        payStatus: json['payStatus'] ?? "",
        receivingType: json['receivingType'] ?? "",
        sum: json['sum'] ?? "");
  }
}
